### ISSUE TITLE
Regression in grep with Junctions

### ISSUE THREAD

**Issue by 0racle** [rakudo/rakudo#2975]
## The Problem

Using an expression that returns a Junction inside `grep` should be evaluated in Boolean context

## Expected Behavior

See below output from old compiler

```perl-6
> $*PERL.compiler
rakudo (2018.04.1)
> ((0, 0), (0, 1), (1, 1)).grep(*.any) 
((0 1) (1 1))
> ((0, 0), (0, 1), (1, 1)).grep(*.one) 
((0 1))
> ((0, 0), (0, 1), (1, 1)).grep(*.none)
((0 0))
```

## Actual Behavior

Output as per later compiler

```perl-6
> $*PERL.compiler
rakudo (2019.03.1.585.gb.8669.fe)
> ((0, 0), (0, 1), (1, 1)).grep(*.any)
()
> ((0, 0), (0, 1), (1, 1)).grep(*.one)
()
> ((0, 0), (0, 1), (1, 1)).grep(*.none)
((0 0) (0 1) (1 1))
```
## Steps to Reproduce

Bisectable points to [this](https://github.com/rakudo/rakudo/commit/c2e272ef682d54758ed1fac80d80e4ee4b1ee0fb) commit

**Comment by lizmat** [comment]
Fixed with 8652386974

**Comment by lizmat** [comment]
Tests added with https://github.com/perl6/roast/commit/f8bef1a604
